### PR TITLE
feat: update ubuntu

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description

Upgrade Open edX image to use Ubuntu ~~24.04~~ 22.04 (See https://github.com/overhangio/tutor/issues/1114#issuecomment-2429082353 for context on 24.04 -> 22.04 switch).

### CODE REVIEW

File: tutor/templates/build/openedx/Dockerfile [hunk 1]
```diff
@@ -138,9 +141,10 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     apt update \
     && apt install -y gettext gfortran graphviz graphviz-dev libffi-dev libfreetype6-dev libgeos-dev libjpeg8-dev liblapack-dev libmysqlclient-dev libpng-dev libsqlite3-dev libxmlsec1-dev lynx mysql-client ntp pkg-config rdfind
 
+
 # From then on, run as unprivileged "app" user
 # Note that this must always be different from root (APP_USER_ID=0)
-ARG APP_USER_ID=1000
+ARG APP_USER_ID=1001
```
Comment (regisb) [inline tutor/templates/build/openedx/Dockerfile:147]: This has pretty big implications. For one, the UID on the host is very often 1000, so changing the internal UID is likely to have more implications in terms of ownership, especially for mounted volumes.  Thus, I think we should avoid running as a different UID. Instead, can we try out this suggestion to delete the existing ubuntu user? https://askubuntu.com/a/1515958

Comment (DawoudSheraz) [reply tutor/templates/build/openedx/Dockerfile:147]: Sure. I initially thought of doing that but it looked weird. Some online suggestions were to use the next available uid which was 1001 in this case.

Comment (DawoudSheraz) [reply tutor/templates/build/openedx/Dockerfile]: I have updated the PR, please take another look.

Comment (DawoudSheraz) [reply tutor/templates/build/openedx/Dockerfile]: @regisb Hi, a reminder on this. Thanks

**Comment by kdmccormick** [comment]
2U found an issue with 24.04 (which I don't yet understand) that led them to pin edx-platform tests back to 22.04: https://github.com/openedx/edx-platform/pull/35635

I'm going to ask them about it tomorrow morning at Maintenance WG. Perhaps we hold off until edx-platform is back to testing on 24.04?

**Comment by DawoudSheraz** [comment]
> 2U found an issue with 24.04 (which I don't yet understand) that led them to pin edx-platform tests back to 22.04: [openedx/edx-platform#35635](https://github.com/openedx/edx-platform/pull/35635)
> 
> I'm going to ask them about it tomorrow morning at Maintenance WG. Perhaps we hold off until edx-platform is back to testing on 24.04?

Sure. @regisb FYI

**Comment by DawoudSheraz** [comment]
@kdmccormick Hi, I have updated the PR to use Ubuntu 22.04. Please take another look. Thanks

**Review by kdmccormick — APPROVED** [review]
Thanks @DawoudSheraz . The diff looks good to me.

I'm going to manually test the build later today when my internet connection is better, but if you have already manually tested it and feel solid about the upgrade, then feel free to merge first.

**Pull request merged**